taichi/analysis/alias_analysis.cpp

```diff
@@ -73,22 +73,15 @@ AliasResult alias_analysis(Stmt *var1, Stmt *var2) {
   if (var1->is<GlobalPtrStmt>() && var2->is<GlobalPtrStmt>()) {
     auto ptr1 = var1->as<GlobalPtrStmt>();
     auto ptr2 = var2->as<GlobalPtrStmt>();
-    bool same = true;
+    bool uncertain = false;
     for (int i = 0; i < (int)ptr1->indices.size(); i++) {
-      if (!irpass::analysis::same_statements(ptr1->indices[i],
-                                             ptr2->indices[i])) {
-        same = false;
-        if (ptr1->indices[i]->is<ConstStmt>() &&
-            ptr2->indices[i]->is<ConstStmt>()) {
-          // different constants
-          return AliasResult::different;
-        }
-      }
-      if (ptr1->indices[i] != ptr2->indices[i]) {
-        same = false;  // TODO: Replace this hotfix with value_diff
-      }
+      int diff = value_diff_ptr_index(ptr1->indices[i], ptr2->indices[i]);
+      if (diff >= (1 << 0))
```

Comment:
Should be `1 << trailing_bits` here... But I'm not sure how to get the trailing bits of the `i`-th index. @yuanming-hu could you please help me with this?

Reply:
Maybe first test if the SNodes of two `ptr`s are the same, return `different` if they aren't; otherwise fetch the common `SNode` as `snode` and what you want would be `snode->extractors[snode->physical_index_position[i]]->trailing_bits`?